### PR TITLE
pipeline(ticdc): stop processing data after sink is stopped

### LINKED ISSUE #5251: stop processing data after sink is stopped
stop processing data after sink is stopped, otherwise sink.Barrier function may hang

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #5251

### What is changed and how it works?
add a flag to identify if the sink is close, if close stop processing data message

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: cdc/processor/pipeline/table_actor.go [hunk 1]
```diff
@@ -62,9 +62,11 @@ type tableActor struct {
 	nodes []*ActorNode
 
 	// states of table actor
-	started  bool
-	stopped  uint32
-	stopLock sync.Mutex
+	started     bool
+	stopped     uint32
+	stopLock    sync.Mutex
```
Comment (hi-rustin) [inline cdc/processor/pipeline/table_actor.go:67]: Is this a lock for sinkNode?

File: cdc/processor/pipeline/table_actor.go [hunk 2]
```diff
@@ -194,12 +196,14 @@ func (t *tableActor) Poll(ctx context.Context, msgs []message.Message[pmessage.M
 		}
 
 		// process message for each node, pull message from parent node and then send it to next node
-		if err := t.handleDataMsg(ctx); err != nil {
-			log.Error("failed to process message, stop table actor ",
-				zap.String("tableName", t.tableName),
-				zap.Int64("tableID", t.tableID), zap.Error(err))
-			t.handleError(err)
-			break
+		if !t.sinkStopped {
```
Comment (hi-rustin) [inline cdc/processor/pipeline/table_actor.go:199]: Is this thread-safe?

**Review by hi-rustin — COMMENTED** [review]
reset LGTM

File: cdc/processor/pipeline/table_actor.go (hunk 1 above)
Comment (sdojjy) [reply cdc/processor/pipeline/table_actor.go:67]: the lock is for actor

File: cdc/processor/pipeline/table_actor.go (hunk 2 above)
Comment (sdojjy) [reply cdc/processor/pipeline/table_actor.go:199]: yes

**Review by asddongmen — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**